src/gf.c

```diff
@@ -1682,18 +1682,41 @@ JL_DLLEXPORT void jl_method_table_disable(jl_methtable_t *mt, jl_method_t *metho
     JL_UNLOCK(&mt->writelock);
 }
 
+static int jl_is_call_ambiguous(jl_value_t *types, jl_method_t *m)
+{
+    if (m->ambig == jl_nothing)
+        return 0;
+    for (size_t i = 0; i < jl_array_len(m->ambig); i++) {
+        jl_typemap_entry_t *mambig = (jl_typemap_entry_t*)jl_array_ptr_ref(m->ambig, i);
+        if (mambig->min_world <= jl_world_counter && jl_world_counter <= mambig->max_world)
+            if (jl_subtype((jl_value_t*)types, (jl_value_t*)mambig->sig))
+                return 1;
+    }
+    return 0;
+}
+
+static int jl_is_function_call_ambiguous(jl_value_t *types JL_PROPAGATES_ROOT, size_t world)
+{
+    jl_methtable_t *mt = jl_first_argument_datatype(types)->name->mt;
+    struct jl_typemap_assoc search = {types, world, NULL, 0, ~(size_t)0};
+    jl_typemap_entry_t *entry = jl_typemap_assoc_by_type(mt->defs, &search, 0, 1);
+    if (!entry)
+        return 0;
+    return jl_is_call_ambiguous(types, entry->func.method);
```

Comment:
It seems like this code should be similar to `jl_gf_invoke_lookup`, but with one of the return values swapped:
```suggestion
    jl_methtable_t *mt = jl_method_table_for(types); // TODO: this should probably just be an argument to this function
    assert((jl_value_t*)mt != jl_nothing);

    struct jl_typemap_assoc search = {(jl_value_t*)types, world, NULL, 0, ~(size_t)0};
    jl_typemap_entry_t *entry = jl_typemap_assoc_by_type(mt->defs, &search, /*offs*/0, /*subtype*/1);
    if (entry == NULL)
        return 0; // we added a new definition to consider
    jl_typemap_entry_t *m = jl_typemap_morespecific_by_type(entry, (jl_value_t*)types, NULL, world);
    if (m == NULL) // TODO: this isn't quite right, since we just ask if the entry dominates over all subtypes of `types`, but wanted to ask the inverse question
        return 1; // a new ambiguity doesn't change the result of method lookup
    return 0; // existing and/or method already covered this intersection (TODO: which is it?)
```
Though I think the `ml_matches_visitor` has the more complete and correct version.